pandas/core/internals/blocks.py

```diff
@@ -1115,11 +1115,8 @@ def check_int_bool(self, inplace):
                                                fill_value=fill_value,
                                                coerce=coerce,
                                                downcast=downcast)
-        # try an interp method
-        try:
-            m = missing.clean_interp_method(method, **kwargs)
-        except ValueError:
-            m = None
+        # validate the interp method
+        m = missing.clean_interp_method(method, **kwargs)
 
```

Comment:
I believe the check `if m is not None` on the following line (1121) and the subsequent raise on line 1132 are now unnecessary, and can be removed.  I'd appreciate any second opinions on that.

Reply:
I'm all for removing it, so long as we are sure that we can't get `None` ever returned for `m`.

Reply:
Great.  `missing.clean_interp_method` will not return None; if its `method` argument is None, it will raise (line 105 of missing.py below):

https://github.com/pandas-dev/pandas/blob/5f73594d49ea06ada4c6779006c00aaf41935595/pandas/core/missing.py#L95-L108